miner/proposal_builder.go

```diff
@@ -68,23 +71,37 @@ type ProposalBuilder struct {
 	tortoise  votesEncoder
 	syncer    system.SyncStateProvider
 
+	mu      sync.Mutex
+	signers map[string]*signerSession
```

Comment:
Why not use `types.NodeID` as the key? It's much easier/more readable to use later and expresses the intent of mapping clearly.
```suggestion
	signers map[types.NodeID]*signerSession
```